### PR TITLE
[hotfix] [build] Disable exclusion rules when using build-jar maven profile

### PR DESCRIPTION
**Pull request by tillrohrmann**
Disables the exclusion rules when using the quickstarts.

### CODE REVIEW

File: flink-quickstart/flink-quickstart-java/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -89,6 +89,61 @@ under the License.
 		</dependency>
 	</dependencies>
 
+	<profiles>
+		<profile>
+			<!-- A profile that does everyting correctly:
```
Comment (zentol) [inline flink-quickstart/flink-quickstart-java/src/main/resources/archetype-resources/pom.xml]: typo: should be everyt**h**ing

Comment (StephanEwen) [inline flink-quickstart/flink-quickstart-java/src/main/resources/archetype-resources/pom.xml]: How about writing `Profile for packaging correct JAR files`. That let's people know it is actually for JAR files ;-)

**Comment by rmetzger** [comment]
Can also also add a note to the documentation about the build-jar profile?


This is just an idea, but maybe we could add the following plugin to the quickstarts:

```xml
			<plugin>
				<artifactId>maven-antrun-plugin</artifactId>
				<executions>
					<execution>
						<phase>generate-resources</phase>
						<goals>
							<goal>run</goal>
						</goals>
						<configuration>
							<tasks>
								<echo>${line.separator}
									${line.separator}Please use the "-Pbuild-jar" profile for creating a fat-jar.
									${line.separator}${line.separator}${line.separator}</echo>
							</tasks>
						</configuration>
					</execution>
				</executions>
			</plugin>
```

this leads to the following output:

```
obert@robert-da ...nk-workdir/flink-state-watermarks-1.0 (git)-[no_kryo] % mvn install

[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Flink Quickstart Job 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-antrun-plugin:1.3:run (default) @ flink-state-watermarks-1.0 ---
[INFO] Executing tasks
     [echo] 
     [echo] 
     [echo] 									
     [echo] Please use the "-Pbuild-jar" profile for creating a fat-jar.
     [echo] 									
     [echo] 
[INFO] Executed tasks
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ flink-state-watermarks-1.0 ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ flink-state-watermarks-1.0 ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 

```

File: flink-quickstart/flink-quickstart-java/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (tillrohrmann) [inline flink-quickstart/flink-quickstart-java/src/main/resources/archetype-resources/pom.xml]: Good catch. Will replace it with @StephanEwen's proposal.

**Comment by tillrohrmann** [comment]
Thanks for the review @zentol, @StephanEwen and @rmetzger. I've changed the comment.

Concerning the documentation, there is already a note about the `build-jar` profile. In fact that's also how we describe to build the fat jar in our [quickstart guide](https://ci.apache.org/projects/flink/flink-docs-master/quickstart/java_api_quickstart.html#build-project). Do you think that it should be put more prominently?

**Comment by tillrohrmann** [comment]
Concerning the `maven-antrun-plugin`, I think it wouldn't hurt to include it. However, it might also simply disappear under all the other maven console output.

**Comment by rmetzger** [comment]
I'm okay with not including it.

**Comment by rmetzger** [comment]
+1 to merge this change.

**Comment by rmetzger** [comment]
Merging ...

**Pull request closed without merging**